AirCasting/CreateSessionViews/CreateSessionDetails/CreateSessionDetailsView.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  AddNameAndTagsView.swift
+//  AirCasting
+//
+//  Created by Anna Olak on 24/02/2021.
+//
+import AirCastingStyling
+import SwiftUI
+
+struct CreateSessionDetailsView: View {
+    
+    @EnvironmentObject private var sessionContext: CreateSessionContext
+    @StateObject var viewModel: CreateSessionDetailsViewModel
+    @Binding var creatingSessionFlowContinues: Bool
+    
+    var body: some View {
+        GeometryReader { geometry in
+            ScrollView(.vertical) {
+                    VStack(alignment: .leading, spacing: 25) {
+                        ProgressView(value: 0.75)
+                        titleLabel
+                        VStack(spacing: 20) {
+                            createTextfield(placeholder: Strings.CreateSessionDetailsView.sessionNamePlaceholder, binding: $viewModel.sessionName)
+                            createTextfield(placeholder: Strings.CreateSessionDetailsView.sessionTagPlaceholder, binding: $viewModel.sessionTags)
+                        }
+                        if sessionContext.sessionType == SessionType.fixed {
+                            placementPicker
+                            transmissionTypePicker
+                            if viewModel.showCompleteCredentials() {
+                                providePasswordTitle
+                                if #available(iOS 15.0, *) {
+                                    createTextfield(placeholder: Strings.WifiPopupView.wifiPlaceholder, binding: $viewModel.wifiSSID)
+                                        .onSubmit { viewModel.isSSIDTextfieldDisplayed = false }
+                                } else {
+                                    createTextfield(placeholder: Strings.WifiPopupView.wifiPlaceholder, binding: $viewModel.wifiSSID)
+                                }
+                                createTextfield(placeholder: Strings.WifiPopupView.passwordPlaceholder, binding: $viewModel.wifiPassword)
+                            } else if viewModel.isWiFi {
+                                provideNameAndPasswordTitle
+                                createTextfield(placeholder: Strings.WifiPopupView.passwordPlaceholder, binding: $viewModel.wifiPassword)
+                                connectToDifferentWifi
+                            }
+                        }
+                        Spacer()
+                        continueButton
+                    }
+                .padding()
+                .frame(maxWidth: .infinity, minHeight: geometry.size.height, alignment: .top)
+            }
+            .background(navigation)
+        }
+        .onAppear { viewModel.onScreenEnter() }
+    }
+}
+
+private extension CreateSessionDetailsView {
+    
+    var navigation: some View {
+        ZStack {
+            NavigationLink(
```

Comment:
let's move both of those NavigationLinks to separate `private vars` with nice names (locationPickerLink and createSesssionLink or smth)